execute/transformation.go

```diff
@@ -87,7 +88,7 @@ func (ts TransformationSet) Finish(id DatasetID, err error) {
 // StreamContext represents necessary context for a single stream of
 // query data.
 type StreamContext interface {
-	Bounds() *Bounds
+	Bounds() *interval.Bounds
```

Comment:
This change isn't clear to me. It seems that `interval.Bounds` is intended to be created from the window function and that the `interval.Bounds` itself isn't meaningful outside of it being generated by the windowing function. I fear that using `interval.Bounds` everywhere where we need a start and stop and allowing a person to create a bounds will invalidate some of the math included in the bounds with the index and give a false sense of security. I wonder if we can keep this the same so `execute.Bounds` is used to denote a complete boundary (such as for range) while `interval.Bounds` is used to describe the boundary of an interval. Does that make sense?

Reply:
Yeah, I guess that makes some sense. A version of bounds that is mutable for creating range bounds and a version of bounds that are created for windowing and immutable. Is that right?

Reply:
I went ahead and took your suggesion here. Only windowing bounds should now be affected.